src/controllers/Product.controller.ts

```diff
@@ -32,10 +36,23 @@ export const searchProducts = async (req: Request, res: Response) => {
       whereClause.name = { [Op.like]: `%${name}%` };
     }
 
+    // if only minPrice is provided or only maxPrice is provided
```

Comment:
Nice implementation here but try wrapping the search logic into a function and a switch case where the function could take params of which query parameter it received and selects that from a switch case and returns the required data according to what was selected. 
```suggestion
    // if only minPrice is provided or only maxPrice is provided
```

Reply:
Thanks a lot for the recommendation, actually,
Been a while since I used a switch statement.
But yeah, did that, and "Switched" the logic to a utility function to break it all down into manageable components.